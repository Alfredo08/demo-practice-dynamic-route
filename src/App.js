import React from 'react';
import {Route, Link, Switch} from 'react-router-dom';
import Drink from './Drink';
import NoMatch from './NoMatch';

class App extends React.Component{
  state = {
    drinks : []
  };

  getDrinks = () => {

    let url = "https://www.thecocktaildb.com/api/json/v1/1/search.php?s=margarita";

    fetch( url )
      .then( response => {
        if( response.ok ){
          return response.json();
        }

        throw new Error( "Something went wrong" );
      })
      .then( responseJSON => {
        this.setState({
          drinks : responseJSON.drinks
        });
      });
  }

  render(){
   return(
    <div>
       <div>
         The drinks api
      </div>
      <button onClick={this.getDrinks}>
        Click to grab drinks
      </button>

      {
        this.state.drinks.map( ( drink, index ) => {
          return (
            <div key={index}>
              <Link to={`/currentDrink/${drink.idDrink}`}> {drink.strDrink}  </Link>
            </div>);
        })
      }
      <Switch>
        <Route exact path="/currentDrink/:currentIdDrink" render={ (routeProps) => {
        
          const currentDrink = this.state.drinks.find( ( drink ) => {
            if( drink.idDrink === routeProps.match.params.currentIdDrink ){
              return drink;
            }
          });

          if( currentDrink ){
            return (<Drink drink={currentDrink} {...routeProps}/>) 
          }
          else{
            return ( <NoMatch error="The id of the drink doesn't exist!"/> );
          }
        }}
        />
        <Route exact path="/" />
        <Route render={() => <NoMatch error="This route doesn't exist!"/> } />
      </Switch>
      
     </div>
   )
  }
}

export default App;