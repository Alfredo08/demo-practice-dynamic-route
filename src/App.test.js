import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import App from './App';

const testHobbies = [{
  name : "Dance"
},
{
  name : "Sing"
}]

// Smoke test
it ('Should load the App component correctly', () => {
  let container = document.createElement( 'div' );
  ReactDOM.render( <App hobbies={testHobbies} />, container );
  ReactDOM.unmountComponentAtNode( container );
});


// Snapshot test
it ( 'Renders the virtual App in the UI correctly', () => {
  let container = renderer.create( <App hobbies={testHobbies} />).toJSON();
  expect( container ).toMatchSnapshot();
});