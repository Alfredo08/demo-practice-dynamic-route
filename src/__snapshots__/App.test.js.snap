// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders the virtual App in the UI correctly 1`] = `
<div>
  <h1>
     List of hobbies 
  </h1>
  <ul>
    <li>
      <div>
        <h2>
          I really like to 
          Dance
        </h2>
      </div>
    </li>
    <li>
      <div>
        <h2>
          I really like to 
          Sing
        </h2>
      </div>
    </li>
  </ul>
</div>
`;
